dvc/remote/index.py

```diff
@@ -0,0 +1,78 @@
+import logging
+import pathlib
+import pickle
+
+logger = logging.getLogger(__name__)
+
+
+class RemoteIndex(object):
+    """Class for locally indexing remote checksums.
+
+    Args:
+        repo: repo for this remote
+    """
+
+    INDEX_SUFFIX = ".idx"
+
+    def __init__(self, repo, name):
+        if repo and hasattr(repo, "index_dir") and name:
+            self.path = pathlib.Path(repo.index_dir).joinpath(
+                "{}{}".format(name, self.INDEX_SUFFIX)
+            )
+        else:
+            self.path = None
+        self._checksums = set()
+        self.load()
+
+    def __iter__(self):
+        return iter(self._checksums)
+
+    @property
+    def checksums(self):
+        return self._checksums
+
+    def load(self):
+        """(Re)load this index from disk."""
+        if self.path and self.path.exists():
+            try:
+                with open(self.path, "rb") as fd:
```

Comment:
```suggestion
                with open(self.path, "rb") as fobj:
```
let's use `fobj`, since `fd` is misleading. We have a mix all over the code, but `fobj` is the preferred one for the new code.